torch/distributed/_spmd/graph_utils.py

```diff
@@ -104,6 +104,20 @@ def get_output(graph: fx.Graph) -> fx.Node:
     raise RuntimeError(f"Cannot find the output node in {graph}")
 
 
+def find_node(
+    graph: fx.Graph, predicate: Callable, reverse_order: bool = False
+) -> List[fx.Node]:
+    """
+    Take a predicate and return all the nodes in the `graph` where the predicate
+    holds.
+    """
+    nodes = graph.nodes
+    if reverse_order:
+        nodes = reversed(nodes)
```

Comment:
nit: simplification?

```python
nodes = reversed(graph.nodes) if reverse_order else graph.nodes
```

Reply:
This is less easier to read after typing hints. Will keep the original form.